modules/dnn/src/int8layers/convolution_layer.cpp

```diff
@@ -1384,10 +1386,36 @@ class ConvolutionLayerInt8Impl CV_FINAL : public BaseConvolutionLayerInt8Impl
         int nstripes = std::max(getNumThreads(), 1);
         Mat outputInt32 = Mat(shape(outputs[0]), CV_32S);
 
-        ParallelConv::run(inputs[0], outputInt32, weightsMat, outputMultiplier, biasvec, activationLUT, kernel_size, strides,
-                          pads_begin, pads_end, dilations, activ.get(), ngroups, nstripes, input_zp, output_zp);
+        // TODO! optimize the fastQConv by AVX/AVX2, need to rethink the Pack_layout.
+        // Currently, on x86 platform, the fastQConv branch is slower than ParallelConv branch.
+        // So we keep it for now.
+#if CV_TRY_AVX2 || CV_TRY_AVX
+        {
+            ParallelConv::run(inputs[0], outputInt32, weightsMat, outputMultiplier, biasvec, activationLUT, kernel_size, strides,
+                              pads_begin, pads_end, dilations, activ.get(), ngroups, nstripes, input_zp, output_zp);
 
-        outputInt32.convertTo(outputs[0], CV_8S);
+            outputInt32.convertTo(outputs[0], CV_8S);
+        }
+#else
+        if (!fastQConvImpl)
+        {
+            int K = outputs[0].size[1];
+            int C = inputs[0].size[1];
+
+            int conv_dim = CONV_2D;
+            if (inputs[0].dims == 3)
+                conv_dim = CONV_1D;
+            if (inputs[0].dims == 5)
+                conv_dim = CONV_3D;
+
+            CV_Assert(outputs[0].size[1] % ngroups == 0);
+            fastQConvImpl = initFastQConv(weightsMat, &biasvec[0], ngroups, K, C, kernel_size, strides, dilations,
+                                          pads_begin, pads_end, conv_dim, outputMultiplier, input_sc, input_zp,
+                                          output_sc, output_zp, per_channel);
+        }
+
+        runFastQConv(inputs[0], outputs[0], fastQConvImpl, nstripes, activ);
+#endif
```

Comment:
Currently, on AVX/AVX2 platforms, the new branch(fastQConv) is slower than the old branch due to the lack of a better strategy. Therefore, the old branch(ParallelConv) is currently kept and I will continue to be optimized in `4.9` and remove the old branch.

Reply:
yes, AVX2-VNNI or AVX512-VNNI extension should be present to make 8-bit convolution efficient on Intel. It should be a separate mini-project to enable support for those instructions at cmake, macro definitions, dispatcher and the kernel level.